COLLADAStreamWriter/src/COLLADASWLibrary.cpp

```diff
@@ -43,5 +45,30 @@ namespace COLLADASW
         mLibraryOpen = false;
     }
 
+    //---------------------------------------------------------------
+    void Library::openExtraLibrary(const String & profile)
+    {
+        if ( !mLibraryOpen )
+        {
+            mExtra.openExtra();
+            mTechnique.openTechnique(profile);
+            mLibraryCloser = mSW->openElement ( mName );
+            mLibraryOpen = true;
+        }
+    }
+
+
+    //---------------------------------------------------------------
+    void Library::closeExtraLibrary()
+    {
+        if (mLibraryOpen)
+        {
+            mLibraryCloser.close();
+            mTechnique.closeTechnique();
+            mExtra.closeExtra();
+        }
+
+        mLibraryOpen = false;
```

Comment:
in this function, libraryOpen=false is done outside of the if() statement
in the function above, libraryOpen=true is done inside the if() statement.

Probably you want to make this line inside the if() statement in both functions?